### ISSUE TITLE
Metacello should be the way to install a project

### ISSUE THREAD

**Issue by jecisc** [SquareBracketAssociates/EnterprisePharo#67]
This is a question:

I think that the best way to load a project is via Metacello. But in some chapter there is Gofer scripts. Should we change them?

Example:
Gofer it
url: 'http://smalltalkhub.com/mc/estebanlm/Voyage/main';
configurationOf: 'VoyageMongo';
loadStable.

should be

Metacello new
	smalltalkhubUser: 'estebanlm' project: 'Voyage';
	configuration: 'VoyageMongo';
	version: #stable;
	load

**Comment by kilon** [comment]
There is a constant sate of flux in this area, you are correct to note that we no longer use Gofer as much as we did in the past. However even metacello is on its way out to an extend because smalltalkhub has been abandonware for a long time now and the goal is to move to git, and mostly github or other websites. 

Thats means versioning will aso be handles by git in the features, through branches, fork, releases, git tags and much more depending on the demands of each project. 

Generally I am not to eager to update this section because I predict that several things are going to change in the process. Already a lot of things has changes, we have tools that integrate already with github to the point you dont even need git installed to make commits, gitfiletree has versions that reject/ignore meta info filetree to move versioning completely on git and much , much more. 

Obviously Pharo being a smalltalk means we will have to find the balance that suits our live coding workflow. So for now there is no best way. 

Actually my best way so far is a combination of git, filetree, Catalog Browser, makefile and startupscript which has completely replaced my use of pharoLauncher sadly and probably will lead me to produce a new GUI tools for building images. 

So for now my advise to people is yes use configurations only if you want to works with StHub and you want to add your project to Catalog Browser. For everyone else git is the way to go, and I dont see why we cant also support mecurial and other VCS . The workflow afterall is pretty much the same.

**Comment by jfabry** [comment]
I actually prefer gofer over metacello. But I am OK with metacello being used as well as gofer, in the book.

I think it is more important to focus on the other open issues, which are more important, and close them. This book has been without progress for too long and if we keep opening issues instead of closing them we will never get it done ...

**Comment by estebanlm** [comment]
but Metacello should be the recommended way to install projects. 
in the “regular” case, is more or less the same, but Metacello has ways to deal with “exceptional” cases (like conflicts, upgrades, etc.)

Esteban

**Comment by jecisc** [comment]
The problem is that they are not the same. The same configuration will bring different packages with Gofer and Metacello.
I heard that Metacello should be the default one (Christophe, Dale…). If it should be the default we should check that all configurations are fine with it and use it. The less we use Gofer to load configuration, the sooner we will have an unified api.
And Metacello can load configurations/baselines from git aswell.
But I agree that it is not the top priority. I just think that we should aim perfection, and this issue will be needed to get it.

**Comment by estebanlm** [comment]
> On 09 Mar 2016, at 14:53, CyrilFerlicot <notifications@github.com> wrote:
> 
> The problem is that they are not the same. The same configuration will bring different packages with Gofer and Metacello.
> 
no it shouldn’t. 
if that is happening, there is a bug somewhere. 

Esteban
> I heard that Metacello should be the default one (Christophe, Dale…). If it should be the default we should check that all configurations are fine with it and use it. The less we use Gofer to load configuration, the sooner we will have an unified api.
> 

as I said… if it does not installs the same, there is an error.

**Comment by jfabry** [comment]
2 things quickly:
- supposedly the install instructions have been tested when the text was written or edited. So the gofer vs metacello bug (that has happened to me in other cases) is not an issue. Worse even, if we change the install instructions the bug can appear and we will lose more time.
- Le mieux est l'ennemi du bien (Perfect is the enemy of good) ;-)

**Comment by jecisc** [comment]
We cannot use Metacello now at Synectique because our configurations works only with gofer.
Our current configuration works with gofer probably because of bugs in gofer (for example if we load two times the same configuration we have some different packages each time).
So yes, they bring different packages. This is due to some bugs in Gofer but they still bring different packages. We should not use these bugs but sometime we don't know we use them and they are hard to correct.

**Comment by kilon** [comment]
I agree for now Metacello is the default and not Gofer. I dont think I have seen people mention Gofer for over a year in the mailing list and I have no idea if its still developed. On the other hand Metacello is currently heavily used.

**Comment by DamienCassou** [comment]
Too late to change that now.

**Comment by tinchodias** [comment]
This is fixed by #118, right?
